.github/workflows/compatibility-check-cross-version.yml

```diff
@@ -0,0 +1,133 @@
+name: Check cross-version compatibility
+
+on:
+  workflow_dispatch:
+    inputs:
+      ccud-version:
+        description: The CCUD version to test. Leave blank to build and test CCUD from sources.
+        required: false
+        type: string
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      dev-ccud-version: ${{ steps.determine-current-version.outputs.ccud-version }}
+      test-ccud-version: ${{ steps.determine-test-version.outputs.ccud-version }}
+    steps:
+      - name: Checkout local repo
+        uses: actions/checkout@v4
+
+      - name: Read current CCUD version from release/version.txt
+        id: determine-current-version
+        run: 
+          echo "ccud-version=$(cat release/version.txt)" >> "$GITHUB_OUTPUT"
+
+      - name: Determine test CCUD version
+        id: determine-test-version
+        run: 
+          echo "ccud-version=${{ inputs.ccud-version || steps.determine-current-version.outputs.ccud-version }}" >> "$GITHUB_OUTPUT"
+
+      - name: Log current CCUD versions
+        run: |
+          echo "dev-ccud-version = ${{ steps.determine-current-version.outputs.ccud-version }}"
+          echo "test-ccud-version = ${{ steps.determine-test-version.outputs.ccud-version }}"
+
+      - name: Set up Java
+        if: ${{ steps.determine-current-version.outputs.ccud-version == steps.determine-test-version.outputs.ccud-version }}
```

Comment:
A very minor thing, but you could create a boolean output in a previous step with the result of `steps.determine-current-version.outputs.ccud-version == steps.determine-test-version.outputs.ccud-version` and replace all of the subsequent occurrences.